Roslyakov Evgeny/hw_js_funcs.js

```diff
@@ -0,0 +1,162 @@
+// ---------- 1 ---------- //
+
+function splitAndMerge(str, sp) {
+    return str.split(' ')
+        .map(function(word) {
+            return word.split('')
+        })
+        .map(function(newWord) {
+            return newWord.join(sp)
+        })
+        .reduce(function(acc, cur) {
+            return acc + ' ' + cur + ' '
+        })
+        .trim();
+}
+
+// console.log(splitAndMerge("My name is John"," "));
+// console.log(splitAndMerge("Hello World!",","));
+
+
+// ---------- 2 ---------- //
+
+function convert(hash) {
+    var res = [];
+        temp = [];
+    for(key in hash) {
+        temp.push(key, hash[key]);
+        res.push(temp);
+        temp = [];
+    }
+
+    return res;
+}
+
+// console.log(convert({name: 'Jeremy', age: 24, role: 'Software Engineer'}));
+
+
+// ---------- 3 ---------- //
+
+function toCamelCase(str) {
+    return str.split('')
+        .reduce(function(acc, cur, i, arr) {
+            if(cur === '-' || cur === '_') {
+                arr[i+1] = arr[i+1].toUpperCase();
+                return acc;
+            }
+            return acc += cur;
+        })
+}
+
+// console.log(toCamelCase("the-stealth-warrior"));
+// console.log(toCamelCase("The_Stealth_Warrior"));
+
+
+// ---------- 4 ---------- //
+
+function reverse(str) {
+    var temp = str.split(''),
+        res = '';
+    for (var i = temp.length - 1; i >= 0; i--) {
+        res += temp[i];
+    }
+    return res;
+}
+
+// console.log(reverse(" A fun little challenge! "));
```

Comment:
" A fun little challenge! " => " A nuf elttil !egnellahc " should be the result of the function. However, your result is " !egnellahc elttil nuf A ".

Reply:
Correct version of this issue is added